src/services/block_service.cpp

```diff
@@ -73,7 +73,7 @@ bool block_service::start()
 void block_service::work()
 {
     zmq::socket xpub(authenticator_, role::extended_publisher, external_);
-    zmq::socket xsub(authenticator_, role::extended_subscriber, internal_);
+    zmq::socket xsub(authenticator_, role::puller, internal_);
 
     // Bind sockets to the service and worker endpoints.
     if (!started(bind(xpub, xsub)))
```

Comment:
The reason for using XPUB is to proxy it against XSUB. I was not aware that XPUB could proxy against anything other than XSUB. I'm not confident we have this right. See [ZeroMQ documentation](http://zguide.zeromq.org/php:chapter2#The-Dynamic-Discovery-Problem).

Reply:
Ah, I think you're right.  zmq usually fails or fails to deliver for unsupported operations, so I'm not sure how they let this slide ;-)  I'll ask about it, and also update this.